geode-wan/src/test/java/org/apache/geode/internal/cache/wan/GatewayReceiverImplJUnitTest.java

```diff
@@ -145,12 +150,17 @@ public void testFailToStartWithARangeOfPorts() throws IOException {
     InternalCache cache = mock(InternalCache.class);
     CacheServerImpl server = mock(CacheServerImpl.class);
     when(cache.addCacheServer(eq(true))).thenReturn(server);
-    doThrow(new SocketException("Address already in use")).when(server).start();
+    AtomicInteger callCount = new AtomicInteger();
+    doAnswer(invocation -> {
```

Comment:
This can actually be simplified using built in Mockito mechanisms for checking call counts:

```
  @Test
  public void testFailToStartWithARangeOfPorts() throws IOException {
    InternalCache cache = mock(InternalCache.class);
    CacheServerImpl server = mock(CacheServerImpl.class);
    when(cache.addCacheServer(eq(true))).thenReturn(server);
    doThrow(new SocketException("Address already in use")).when(server).start();
    GatewayReceiverImpl gateway =
        new GatewayReceiverImpl(cache, 2000, 2100, 5, 100, null, null, null, true);
    assertThatThrownBy(() -> gateway.start()).isInstanceOf(GatewayReceiverException.class)
        .hasMessageContaining("No available free port found in the given range");
    assertTrue(gateway.getPort() == 0);
    verify(server, times(101)).start();
  }
```